api/client/login.go

```diff
@@ -74,6 +74,10 @@ func (cli *DockerCli) CmdLogin(args ...string) error {
 			username = authconfig.Username
 		}
 	}
+
+	if username == "" {
+		return fmt.Errorf("Error: Non-null Username Required")
+	}
```

Comment:
in this case and above (line 69) you should `username = strings.TrimSpace(username)` first just to make sure people don't give you a list of spaces.